tests/forms_tests/field_tests/test_jsonfield.py

```diff
@@ -0,0 +1,110 @@
+import json
+import uuid
+
+from django import forms
+from django.core import exceptions
+from django.core.serializers.json import DjangoJSONEncoder
+from django.test import SimpleTestCase
+
+
+class CustomDecoder(json.JSONDecoder):
+    def __init__(self, object_hook=None, *args, **kwargs):
+        return super().__init__(object_hook=self.object_hook, *args, **kwargs)
+
+    def object_hook(self, dct):
+        try:
+            dct['uuid'] = uuid.UUID(dct['uuid'])
+        except KeyError:
+            pass
+        return dct
+
+
+class TestFormField(SimpleTestCase):
+
+    def test_valid(self):
+        field = forms.JSONField()
+        value = field.clean('{"a": "b"}')
+        self.assertEqual(value, {'a': 'b'})
+
+    def test_valid_empty(self):
+        field = forms.JSONField(required=False)
+        value = field.clean('')
+        self.assertIsNone(value)
+
+    def test_invalid(self):
+        field = forms.JSONField()
+        with self.assertRaisesMessage(exceptions.ValidationError, 'Enter a valid JSON value.'):
+            field.clean('{some badly formed: json}')
+
+    def test_formfield_disabled(self):
+        class JsonForm(forms.Form):
+            name = forms.CharField()
+            jfield = forms.JSONField(disabled=True)
+
+        form = JsonForm({'name': 'xyz', 'jfield': '["bar"]'}, initial={'jfield': ['foo']})
+        self.assertIn('[&quot;foo&quot;]</textarea>', form.as_p())
+
+    def test_prepare_value(self):
+        field = forms.JSONField()
+        self.assertEqual(field.prepare_value({'a': 'b'}), '{"a": "b"}')
+        self.assertEqual(field.prepare_value(None), 'null')
```

Comment:
Is this the behaviour on the existing `django.contrib.postgres` JSONField?

Reply:
Yes.

Reply:
Refer to https://github.com/django/django/blob/master/tests/postgres_tests/test_json.py#L527